### ISSUE TITLE
find / with rm -rf does produce warning 

### ISSUE THREAD

**Issue by harish2704** [koalaman/shellcheck#2328]
#### For bugs
- My shellcheck version : online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash


find / -exec rm -rf {} \;
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
No issues detected!

```
#### Here's what I wanted or expected to see:

The above command is equivalent to `rm -rf /` so it should be treated in that way. Thus, I was expecting a warning message

**Comment by koalaman** [comment]
You're right, ideally such cases would be caught. In this particular case it requires modelling `find`'s behavior and realizing that being invoked with a path `/` and no further predicates will cause the -exec to run with `{}` to be substituted for `/` and `/usr` and several other directories that probably shouldn't be removed. 

This type of symbolic execution is unfortunately beyond ShellCheck at this point, and likely always will be.

The `rm -rf /` warning does not aim to prove that a script is safe and won't delete files, but much less ambitiously to help detect typo accidents like the infamous `rm -rf /usr /lib/nvidia-current/xorg/xorg` case.
